### ISSUE TITLE
Make Model implement MutableMapping

### ISSUE THREAD

**Issue by kjaquier** [schematics/schematics#396]
The `Model` class has all the methods defined in [collections.abc.MutableMapping](https://docs.python.org/3.5/library/collections.abc.html), but doesn't actually implement it:
```python
>>> from collections.abc import *
>>> from schematics.models import Model
>>> class Foo(Model): pass
... 
>>> mutable_mapping_attrs = ('__getitem__', '__setitem__', '__delitem__', '__iter__', '__len__')
>>> all(hasattr(Foo(), attr) for attr in mutable_mapping_attrs)
True
>>> isinstance(Foo(), MutableMapping)
False
```

Adding `MutableMapping` to the base classes of `Model` would provide automatic implementations for usual dict methods such as `update`, which would be quite nice since one would expect a `Model`'s instance to behave exactly like a dict since it "looks like it and quacks like it".

**Comment by bintoro** [comment]
I agree in principle, but this is more complicated than it seems. The question is being discussed in #375. Please have a look, particularly the last few comments. Any input is more than welcome!

In short, my current thinking is that it might make sense to actually *remove* some methods by default but provide a mixin that would add the full mapping interface as you suggest.
